backend-rust/src/graphql_api.rs

```diff
@@ -693,21 +693,25 @@ impl BaseQuery {
             -- to select page items from the start/end of the range.
             -- Each page must still independently be ordered ascending.
             -- See also https://relay.dev/graphql/connections.htm#sec-Edge-order
-            ORDER BY CASE
-                WHEN $3 THEN index
-                WHEN $4 THEN amount
-                WHEN $5 THEN num_txs
-                WHEN $6 THEN delegated_stake
-            END ASC",
+            ORDER BY
```

Comment:
Ehh. Dont we just double sort this?

Reply:
Some of these are redundant, yes, but I chose not to special case these as the query is already plenty complex.

Reply:
I was hoping for something to dynamically apply `ASC or DESC` when the query parameter is inserted to avoid some of these cases. But haven't found support for it yet :(.

Looking for something like this:

```
let mut accounts = sqlx::query_as!(
    Account,
    r"
    
    .....     

    ORDER BY
        CASE WHEN $3 THEN index END $10,
        CASE WHEN $4 THEN amount END $10,
        CASE WHEN $5 THEN num_txs  END $10,
        CASE WHEN $6 THEN delegated_stake END $10
    ",
    query.from,
    query.to,
    
    ....
    
    if matches!(order.dir, OrderDir::Desc) { "DESC" } else { "ASC" },
)
.fetch(pool);
```

Reply:
Since these will be prepared statements, it is not possible to change keywords dynamically, only the query parameters can be dynamic, so we have to do something like this.